File: bank-service/src/main/java/rs/edu/raf/model/credit/CreditRequest.java
package rs.edu.raf.model.credit;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import rs.edu.raf.model.Currency;
import rs.edu.raf.model.accounts.BankAccount;

import java.math.BigDecimal;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class CreditRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long clientId;
    @ManyToOne
    private CreditType creditType;
    private BigDecimal loanAmount;
    @ManyToOne
    private Currency currency;
    private String loanPurpose;
    private BigDecimal salary;
    private String phoneNumber;
    private boolean permanentEmployee;
    private int currentEmploymentPeriod;
    private int loanTerm;
    @ManyToOne
    private BankAccount bankAccount;
    @Enumerated
    private CreditRequestStatus creditRequestStatus;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true, mappedBy = "creditRequest")
    private List<Credit> credits;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/credit/CreditRequest.kt
package org.example.bankservicekotlin.model.credit

import jakarta.persistence.*
import org.example.bankservicekotlin.model.accounts.BankAccount
import java.math.BigDecimal

@Entity
class CreditRequest(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    var clientId: Long?,
    @ManyToOne
    val creditType: CreditType,
    val loanAmount: BigDecimal,
    val loanPurpose: String,
    val salary: BigDecimal,
    val phoneNumber: String,
    val permanentEmployee: Boolean,
    val currentEmploymentPeriod: Int,
    val loanTerm: Long,
    @ManyToOne
    val bankAccount: BankAccount,
    @Enumerated
    var creditRequestStatus: CreditRequestStatus,
    @OneToMany(cascade = [CascadeType.ALL], fetch = FetchType.LAZY, orphanRemoval = true, mappedBy = "creditRequest")
    val credits: MutableList<Credit>?
) {
}